### PR TITLE
fix: table widget issue where editableCell is undefined

### LINKED ISSUE #16725: [Bug]: Table widget crashes when editableCell is undefined on page load
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In deploy mode, if the Table widget editable cell value is undefined, the whole application crashes.
https://discord.com/channels/725602949748752515/760761686549463060/1018991583057682484

### Steps To Reproduce

Unable to repro on my end. Only happens on the user's application.
refer https://discord.com/channels/725602949748752515/760761686549463060/1018991583057682484

### Public Sample App

_No response_

### Version

self hosted

### PR DESCRIPTION
**Pull request by sbalaji1192**
Fixes #16725

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -1640,9 +1640,10 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
       value: value,
       inputValue,
     });
+
     this.props.updateWidgetMetaProperty("columnEditableCellValue", {
       ...this.props.columnEditableCellValue,
-      [this.props.editableCell.column]: value,
+      [this.props.editableCell?.column || ""]: value,
```
Comment (arunvjn) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:1646]: Are there any downside to setting the default key to "" ?

**Review by arunvjn — DISMISSED** [review]

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx (hunk 1 above)
Comment (sbalaji1192) [reply app/client/src/widgets/TableWidgetV2/widget/index.tsx:1646]: There won't any downside. `this.props.editableCell?.column` will be present when this function gets executed.

**Pull request merged**